todo/src/repo/repo.ts

```diff
@@ -0,0 +1,109 @@
+import { Task } from "../todo";
+
+interface reposotoryPattern<T> {
+    find(findTodo: String),
+    findAll(),
+    create(makeATodo: T), 
+    del(delTask: String),
+    updateTodo(updateTask: String, payload: T),
+    incompleteTodos(),
+    completeTodos(),
+    orderDates(),
+}
+
+class TodoRepo implements reposotoryPattern<Task> {
+
+    private task: Task[] = [];
+
+    // GET all tasks
+    public findAll() {
+        return this.task
+    }
+
+    // POST
+    public create(newTask) {
+        this.task.push(newTask);
+        return newTask
+    }
+
+    // GET a single task
+    public find(uuid) {
+        const found = this.task.filter(tasks => tasks.uuid == uuid)
+        if(found) {
+            return(found);
+        } else {
+            return (`No task with the ID of ${uuid}`)
+        } 
+    }
+
+    // DEL
+    public del(uuid) {
+        //Find element by uuid
+        const result = this.task.filter(tasks => tasks.uuid == uuid);
+        
+        if(result.length === 0) {
+            return `${uuid} does not exist`
+        }
+
+        const element = result[0]
+
+        // find index of uuid
+        const index = this.task.indexOf(element)
+
+        if (index < 0) {
+            return
+        }
+
+        // splice the array at the indexOf
+        this.task.splice(index, 1);
+        return this.task
+
+    }
+
+    // Update PUT
+    public updateTodo(uuid, payload) {
+        const updateTodo = this.task.filter(tasks => tasks.uuid == uuid);
+
+        if(updateTodo.length === 0) {
+            return `${uuid} does not exist`;
+        }
+        this.task.forEach(tasks => {
+            if (tasks.uuid == uuid) {
+                tasks.todo = payload['todo'];
+                tasks.dueDate = payload['dueDate'];
+                tasks.complete = payload['completed'];
+            }
+        }) 
+        return updateTodo
```

Comment:
This requires 2 loops through the collection which is `O(N^2)`

You can do it with 1 by using the filter and then just mutating the result as the element is still contained in the array even if you assign it to a variable.